### DISCUSSION TITLE
Why handles lighthouse the bounding box for a screenshot differently than my own implementation (fixed elements)

### DISCUSSION THREAD

**Discussion by BennyAlex** [GoogleChrome/lighthouse#15688] [Q&A]
Hello,

I have a problem that some bounding boxes for screenshots in my own implementation doesnt match the preview of the original lighthouse viewer.

As far as I can tell, this only happens for fixed elements. I looked in the source code and also in the report, but I cant find some special case handler for fixed elements.

```
{
"subItems": {
"items": [
{
"relatedNode": {
"boundingRect": {
"right": 1310,
"width": 156,
"left": 1154,
"height": 39,
"top": 901,
"bottom": 940
},
"selector": "body.home > aside.country-selector",
"path": "1,HTML,1,BODY,90,ASIDE",
"type": "node",
"lhId": "1-23-ASIDE",
"nodeLabel": "English",
"snippet": "<aside data-wg-notranslate=\"\" class=\"country-selector weglot-dropdown weglot-default weglot-invert\" tabindex=\"0\" aria-expanded=\"false\" role=\"listbox\" aria-activedescendant=\"weglot-language-en\" aria-label=\"Language selected: English\">"
}
}
],
"type": "subitems"
}
```

The screenshot has a a resolution of 1350x13608. 

My app shows this:
<img width="819" alt="image" src="https://github.com/GoogleChrome/lighthouse/assets/17402724/4415b9fa-a311-4479-94ad-4cbd26dadf27">

Lighthouse this:
<img width="779" alt="image" src="https://github.com/GoogleChrome/lighthouse/assets/17402724/87406095-87c8-4918-9dcb-e6f2722c975e">

So my app shows the bounding box on the correct position (pixel wise), but since its a fixed element, lighthouse shows it at the very bottom of the fullpage screenshot.
Can someone please explain me this and where the code for those elements is, please?

Thanks very much!

**Comment by adamraine** [comment]
Can you provide an example that produces this result?

**Comment by BennyAlex** [comment]
@adamraine 
Hello, please see this repo: https://github.com/BennyAlex/lighthouse-preview-test
Live: https://stackblitz.com/~/github.com/BennyAlex/lighthouse-preview-test


Here is how it looks in the original report: 
<img width="674" alt="image" src="https://github.com/GoogleChrome/lighthouse/assets/17402724/8bef207f-9261-4478-94be-290a319b661c">

<img width="702" alt="image" src="https://github.com/GoogleChrome/lighthouse/assets/17402724/62e3685a-b3e1-4e31-b151-fc56fac2525f">

The item that should be visible is covered by the cookie banner, maybe this is a bug of lighthouse?
But the box is also in another position than its coordinates from the json are.

Here is how it looks using my code:

<img width="463" alt="image" src="https://github.com/GoogleChrome/lighthouse/assets/17402724/e26a72b5-2688-4787-b93c-d73bab740f74">

<img width="461" alt="image" src="https://github.com/GoogleChrome/lighthouse/assets/17402724/43a53685-f24e-449d-9ee2-e32a6d6aeda3">

**Reply by BennyAlex** [reply]
Reported the site https://eye-able.com/en/vision-digital-accessibility/

**Reply by adamraine** [reply]
> The item that should be visible is covered by the cookie banner, maybe this is a bug of lighthouse?

This is what's happening which is the intended behavior. The box highlights the correct position of the element but it is visually blocked by the cookie banner.

**Reply by BennyAlex** [reply]
Yeah sure.

But it still doesn't match the position on the image.
The lighthouse viewer does something different than my one does for certain boxes. What is it? :)

**Reply by BennyAlex** [reply]
Lighthouse places the box to the very bottom of the full page screen shot, while the report box location is mich further at the top.
Since it's an absolute positioned element, I think the bound box coordinates are from a single viewport and not the full page. But why is the lighthouse viewer display the box at the very bottom?

It's either a bug or it can somehow indicate that it is a fixed element that is fixed to the bottom of the viewport and then moves it to the very bottom of the full page screenshot

**Reply by adamraine** [reply]
The positions of the element boxes are the coordinates that element was in on the page, not necessarily the position of the element within the image asset itself. This has led to some confusion in the past (See https://github.com/GoogleChrome/lighthouse/issues/14824).

If you're interested in how the Lighthouse report handles the box positions, the relevant code can be found here:
https://github.com/GoogleChrome/lighthouse/blob/main/report/renderer/element-screenshot-renderer.js#L221

**Reply by BennyAlex** [reply]
Thanks for your help.
I still can figure out the difference between my code and lighthouse, sadly.

Most of the boxes already work with my code, so why do these boxes not work?

In your linked issue I can only see a problem about mismatching image sizes, but not for mismatching bounding box locations.

I don't understand why the coordinates on the page should be different than the coordinates of the screenshot. It should be the same? If not, than the previews wouldn't work I guess.
Also If it is not be the same for some items, then there should be an indicator somewhere, but I can't see that as well.

Do would you kindly explain where the magic happens, please ?

**Reply by BennyAlex** [reply]
@adamraine hey, could you kindly help me with this? :D
